torchvision/prototype/datasets/datapipes.py

```diff
@@ -0,0 +1,29 @@
+from typing import Tuple, IO, Iterator, Union, cast
+
+import torch
+from torchdata.datapipes.iter import IterDataPipe
+
+__all__ = ["FloReader"]
+
+
+class FloReader(IterDataPipe[torch.Tensor]):
+    def __init__(self, datapipe: IterDataPipe[Tuple[str, IO]]) -> None:
+        self.datapipe = datapipe
+
+    def _read_data(self, file: IO, *, dtype: torch.dtype, count: int) -> torch.Tensor:
+        num_bytes_per_value = (torch.finfo if dtype.is_floating_point else torch.iinfo)(dtype).bits // 8
+        chunk_size = count * num_bytes_per_value
+        return torch.frombuffer(bytearray(file.read(chunk_size)), dtype=dtype)
```

Comment:
The value are encoded as little endian in the file - will it be properly decoded on big endian architectures
?

Reply:
No, I need to handle that. Thanks for the heads-up!

Reply:
BTW IDK if this is intended but we decided in the original PR that these would be numpy array
I assume eveything is supposed to be a tensor in datapipes?